### DISCUSSION TITLE
Logstash Fails to Connect to Elasticsearch

### DISCUSSION THREAD

**Discussion by molfar101** [Security-Onion-Solutions/securityonion#8495] [Unsupported Versions]
Hi all,

I've upgraded SO to the latest version (distributed architecture) and the logs don't show up in Kibana. Redis queue is backed up and Logstash on my search nodes fails to connect to Elasticsearch. Here is the error from one of my search nodes. 

[logstash.outputs.elasticsearch] Attempted to resurrect connection to dead ES instance, but got an error {:url=>"https://[IP REMOVED]:9200/", :exception=>LogStash::Outputs::ElasticSearch::HttpClient::Pool::BadResponseCodeError, :message=>"Got response code '401' contacting Elasticsearch at URL 'https://[IP REMOVED]:9200/'"}

I googled and it looks like an authentication issue. I'm not sure how to solve this problem. Please advise.

**Comment by dougburks** [comment]
Did you follow all steps at https://docs.securityonion.net/en/2.3/soup.html#elastic-8 including the following?
> If your deployment was originally installed with a version lower than 2.3.60 and you never enabled Elastic Auth, then you will need to enable Elastic Auth via [so-elastic-auth](https://docs.securityonion.net/en/2.3/so-elastic-auth.html#so-elastic-auth) before updating.

**Reply by molfar101** [reply]
Well the problem is we did enable Elastic Auth but after the upgrade. You think its causing issues because of that? If so, any recommendations how to fix it? I appreciate your help.

**Reply by dougburks** [reply]
It's generally easier if you enable Elastic Auth before the upgrade and that's why we documented it that way.

In theory, you should be able to enable Elastic Auth after the upgrade, but you may need to restart some services.

**Reply by molfar101** [reply]
So I restarted Kibana, Logstash, Redis, and the entire system, but I still have this issue. I have no idea what might be causing it. Do you have any suggestions? Thank you

**Reply by dougburks** [reply]
When you ran `so-elastic-auth`, did it complete successfully?

Are you able to log into Kibana using your SOC username and password?
